core/src/main/java/com/linecorp/armeria/client/retry/RetryingClient.java

```diff
@@ -65,4 +81,54 @@ protected Backoff newBackoff() {
     protected RetryStrategy<I, O> retryStrategy() {
         return retryStrategy;
     }
+
+    /**
+     * Returns the number of requests sent and increase its count by one.
+     */
+    protected int numRequestsSent(ClientRequestContext ctx) {
+        if (originalResponseTimeout == 0) {
+            originalResponseTimeout = ctx.responseTimeoutMillis();
+        }
+        lastTimeoutAndTimestamp.put(numRequestsSent,
+                                    new LastTimeoutAndTimestamp(originalResponseTimeout,
+                                                                Duration.ofNanos(System.nanoTime())
+                                                                        .toMillis()));
+        return numRequestsSent++;
+    }
+
+    /**
+     * Resets the {@link ClientRequestContext#responseTimeoutMillis()}.
+     */
+    protected long resetResponseTimeout(ClientRequestContext ctx, final int reqId) {
+        final long responseTimeoutMillis = responseTimeoutMillis(reqId);
+        if (responseTimeoutMillis <= 0) {
+            throw ResponseTimeoutException.get();
+        }
+
+        lastTimeoutAndTimestamp.get(reqId).lastTimeoutMillis = responseTimeoutMillis;
+        lastTimeoutAndTimestamp.get(reqId).lastTimestampMills = Duration.ofNanos(System.nanoTime()).toMillis();
```

Comment:
I thought I have to use the Duration even though I really didn't want. I was confused. Thank you!